sources/BringgTrackingClient.m

```diff
@@ -149,22 +149,89 @@ - (nullable GGCustomer *)signedInCustomer{
 
 
 - (void)startWatchingOrderWithUUID:(NSString *_Nonnull)uuid
-                        sharedUUID:(NSString *_Nullable)shareduuid
+                        sharedUUID:(NSString *_Nonnull)shareduuid
+                          delegate:(id <OrderDelegate> _Nullable)delegate{
+    
+    NSLog(@"Trying to start watching on order uuid: %@, shared %@, with delegate %@", uuid, shareduuid, delegate);
+    
+    if ([NSString isStringEmpty:uuid] || [NSString isStringEmpty:shareduuid]) {
+        [NSException raise:@"Invalid params" format:@"Order and Share UUIDs can not be empty"];
+        
+        return;
+    }
+    
+    [self.trackerManager startWatchingOrderWithUUID:uuid accessControlParamKey:PARAM_SHARE_UUID accessControlParamValue:shareduuid delegate:delegate];
+}
+
+- (void)startWatchingOrderWithUUID:(NSString *_Nonnull)uuid
+               customerAccessToken:(NSString *_Nonnull)customerAccessToken
                           delegate:(id <OrderDelegate> _Nullable)delegate{
+ 
+    NSLog(@"Trying to start watching using customer token on order uuid: %@, with delegate %@", uuid, delegate);
+    
+    
+    if ([NSString isStringEmpty:uuid] || [NSString isStringEmpty:customerAccessToken]) {
+        [NSException raise:@"Invalid params" format:@"Order and customer token can not be empty"];
+        
+        return;
+    }
     
-    [self.trackerManager startWatchingOrderWithUUID:uuid sharedUUID:shareduuid delegate:delegate];
+    [self.trackerManager startWatchingOrderWithUUID:uuid accessControlParamKey:PARAM_ACCESS_TOKEN accessControlParamValue:customerAccessToken delegate:delegate];
 }
 
 
 
 - (void)startWatchingDriverWithUUID:(NSString *_Nonnull)uuid
-                          shareUUID:(NSString *_Nonnull)shareUUID
```

Comment:
I see that you sometimes write shareUUID and sometimes share**d**UUID so be consistent (with android also)